packages/amplication-server/src/core/subscription/subscription.service.ts

```diff
@@ -199,26 +211,41 @@ export class SubscriptionService {
     });
   }
 
+  async trackUpgradeCompletedEvent(
+    plan: EnumSubscriptionPlan,
+    userId: string,
+    workspaceId: string
+  ): Promise<void> {
+    if (plan !== EnumSubscriptionPlan.Free && userId) {
+      await this.analyticsService.track({
+        userId: userId,
+        properties: {
+          workspaceId: workspaceId,
+        },
+        event: EnumEventType.WorkspacePlanUpgradeCompleted,
+      });
+    }
+  }
+
   async handleUpdateSubscriptionStatusEvent(
     updateStatusDto: UpdateStatusDto
   ): Promise<void> {
     switch (updateStatusDto.type) {
       case "subscription.created": {
         const createSubscriptionInput =
           this.mapUpdateStatusDtoToCreateSubscriptionInput(updateStatusDto);
-        await this.create(updateStatusDto.id, createSubscriptionInput);
-        const userId = updateStatusDto.metadata?.userId;
-        if (
-          createSubscriptionInput.plan !== EnumSubscriptionPlan.Free &&
-          userId
-        ) {
-          await this.analyticsService.track({
-            userId: userId,
-            properties: {
-              workspaceId: updateStatusDto.customer.id,
-            },
-            event: EnumEventType.WorkspacePlanUpgradeCompleted,
-          });
+        const subs = await this.getSubscriptionById(updateStatusDto.id);
+        if (subs) {
```

Comment:
in case a subscription already exists, I think we should update the details instead of ignoring the event

Reply:
this line is under "subscription.created" case, which mean that notification about creation was repeated, it's not an update

Reply:
i just assumed that we if get a notification and the identifier already exist we better update the date then ignore it.
in case this is a duplicate notification - I understand your point. 
but can we guarantee that we will not get this notification for existing users? for example, when a user upgrade from one tier to another, could it be that we get a new notification?